qiskit/transpiler/passes/scheduling/alap.py

```diff
@@ -68,71 +47,89 @@ def run(self, dag):
-            start_time = max(
-                itertools.chain(
-                    (qubit_time_available[q] for q in node.qargs),
-                    (clbit_time_available[c] - delta for c in node.cargs + node.op.condition_bits),
-                )
-            )
-
-            pad_with_delays(node.qargs, until=start_time, unit=time_unit)
+            op_duration = self._get_node_duration(node, bit_indices, dag)
+
+            # compute t0, t1: instruction interval, note that
+            # t0: start time of instruction
+            # t1: end time of instruction
+
+            # since this is alap scheduling, node is scheduled in reversed topological ordering
+            # and nodes are packed from the very end of the circuit.
+            # the physical meaning of t0 and t1 is flipped here.
+            if isinstance(node.op, Measure):
+                # clbit time is always right (alap) justified
+                t0 = max(idle_before[bit] for bit in node.qargs + node.cargs)
+                t1 = t0 + op_duration
+                #
+                #        |t1 = t0 + duration
+                # Q ░░░░░▒▒▒▒▒▒▒▒▒▒▒
+                # C ░░░░░░░░░▒▒▒▒▒▒▒
+                #            |t0 + (duration - clbit_write_latency)
+                #
+                for clbit in node.cargs:
+                    idle_before[clbit] = t0 + (op_duration - self.clbit_write_latency)
+            elif isinstance(node.op, Gate):
+                t0q = max(idle_before[q] for q in node.qargs)
+                if node.op.condition_bits:
+                    # conditional is bit tricky due to conditional_latency
+                    t0c = max(idle_before[c] for c in node.op.condition_bits)
+                    # Assume following case (t0c > t0q):
+                    #
+                    #                |t0q
+                    # Q ░░░░░░░░░░░░░▒▒▒
+                    # C ░░░░░░░░▒▒▒▒▒▒▒▒
+                    #           |t0c
+                    #
+                    # In this case, there is no actual clbit read before gate.
+                    #
+                    #             |t0q' = t0c - conditional_latency
+                    # Q ░░░░░░░░▒▒▒░░▒▒▒
+                    # C ░░░░░░▒▒▒▒▒▒▒▒▒▒
+                    #         |t1c' = t0c + conditional_latency
+                    #
+                    # rather than naively doing
+                    #
+                    #        |t1q' = t0c + duration
+                    # Q ░░░░░▒▒▒░░░░░▒▒▒
+                    # C ░░▒▒░░░░▒▒▒▒▒▒▒▒
+                    #     |t1c' = t0c + duration + conditional_latency
+                    #
+                    t0 = max(t0q, t0c - op_duration)
+                    t1 = t0 + op_duration
+                    for clbit in node.op.condition_bits:
+                        idle_before[clbit] = t1 + self.conditional_latency
+                else:
+                    t0 = t0q
+                    t1 = t0 + op_duration
+            else:
```

Comment:
Conditional measures or unknown non-gate instructions may come here. Why don't we
```
if node.op.condition_bits:
    raise TranspilerError(f"Conditional instruction {node.op.name} is not supported in the ALAP scheduling pass")
```
here? (I know they hadn't been properly handled since the old implementation, though)

Reply:
Good idea!